# ***TURMA: 1CCPO***

## ***Integrantes***

*   Carlos Eduardo Affonso - RM 569676
*   Gabriel Oliveira Gusmão Florêncio dos Santos - RM 573747
*   Gustavo de Souza Abreu - RM 574080
*   Gabrieli de Lima Pettena de Oliveira - RM 569799
*   Igor Massone Monteiro - RM 573853
*   Temitope Kuku da Silva Ogunbanjo - RM 573772


# **Dataset1 - Appliances Energy Prediction (UCI)**

**Fonte**: https://archive.ics.uci.edu/dataset/374/appliances+energy+prediction

**Situação:**

**a empresa de eficiência energética está analisando o comportamento de uma residência de baixo consumo. A equipe deseja identificar períodos de consumo elevado dos eletrodomésticos e observar quais condições de temperatura e umidade estavam presentes nesses momentos.**

In [ ]:
import pandas as pd
dados = pd.read_csv("https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET1.csv")

1. **Carregue a amostra no Pandas e apresente head(), shape, info() e describe().**

In [ ]:
dados.head()

,Appliances,lights,T1,RH_1,T2,RH_2,T3,RH_3
0,40,0,20.8900,35.4000,17.7600,39.1633,20.2900,36.9000
1,90,10,21.8900,53.1000,21.2900,45.3600,21.6333,49.2267
2,50,0,21.3900,35.5000,17.6333,40.5300,21.6667,35.2000
3,50,0,21.3900,41.0333,23.8900,34.8400,22.0333,36.9333
4,70,0,19.9633,35.1267,16.4633,40.1267,20.0000,36.4000


In [ ]:
print(f'Total de registros: {dados.shape[0]}')
print(f'Total de colunas: {dados.shape[1]}')

Total de registros: 1974
Total de colunas: 8


In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1974 entries, 0 to 1973
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Appliances  1974 non-null   int64  
 1   lights      1974 non-null   int64  
 2   T1          1974 non-null   float64
 3   RH_1        1974 non-null   float64
 4   T2          1974 non-null   float64
 5   RH_2        1974 non-null   float64
 6   T3          1974 non-null   float64
 7   RH_3        1974 non-null   float64
dtypes: float64(6), int64(2)
memory usage: 123.5 KB


In [ ]:
dados.describe()

,Appliances,lights,T1,RH_1,T2,RH_2,T3,RH_3
count,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000,1974.000000
mean,93.044580,3.839919,21.664703,40.219193,20.293635,40.411386,22.231615,39.209733
std,93.826149,7.937076,1.574883,3.971923,2.126958,4.037686,1.957275,3.288818
min,20.000000,0.000000,16.790000,27.733300,16.100000,24.063300,17.200000,30.133300
25%,50.000000,0.000000,20.700000,37.205825,18.790000,37.760000,20.700000,36.900000
50%,60.000000,0.000000,21.600000,39.590000,20.000000,40.433300,22.100000,38.433300
75%,100.000000,0.000000,22.600000,42.991675,21.500000,43.228325,23.290000,41.741250
max,770.000000,50.000000,26.260000,57.496700,28.917500,56.026700,29.198600,49.226700


2. **Renomeie Appliances para Consumo_Eletrodomesticos e simplifique o nome de pelo menos três atributos ambientais.**

In [ ]:
dados = dados.rename({'Appliances': 'Consumo_eletrodomesticos',
             'lights':'Luminosidade',
              'T1':'Temperatura_1',
              'RH_1': 'Umidade_1',
              'T2':'Temperatura_2',
              'RH_2':'Umidade_2',
              'T3':'Temperatura_3',
              'RH_3':'Umidade_3' }, axis=1)

print(dados.columns)

Index(['Consumo_eletrodomesticos', 'Luminosidade', 'Temperatura_1',
       'Umidade_1', 'Temperatura_2', 'Umidade_2', 'Temperatura_3',
       'Umidade_3'],
      dtype='object')


3. **Determine o maior consumo de eletrodomésticos registrado na amostra.**

In [ ]:
CMAX = dados ["Consumo_eletrodomesticos"].max()
print(f'O consumm máximo de eletrodomésticos foi de {CMAX:.2f} KW')

O consumm máximo de eletrodomésticos foi de 770.00 KW


4. **Calcule um limiar correspondente a 70% do valor máximo e crie um DataFrame contendo somente os registros acima desse limite.**

In [ ]:
df_alto_consumo = dados[dados['Consumo_eletrodomesticos'] > CMAX * 0.70]
print(f'O limiar correspondente a 70% é de {(CMAX * 0.70):.2f} KW')

O limiar correspondente a 70% é de 539.00 KW


5. **Conte os registros selecionados e calcule qual percentual da amostra eles representam.**

In [ ]:
qtd_registros = len(df_alto_consumo)
percentual = (qtd_registros / len(dados)) * 100

print(f"Registros com consumo acima de 70%: {qtd_registros}")
print(f"Percentual da amostra: {percentual:.2f}%")

Registros com consumo acima de 70%: 22
Percentual da amostra: 1.11%


6. **Calcule a temperatura média de T1 e crie outro DataFrame contendo simultaneamente consumo acima de 70% do máximo e temperatura acima da média.**

In [ ]:
temp_media_t1 = dados['Temperatura_1'].mean()
df_consumo_e_temp = dados[
    (dados['Consumo_eletrodomesticos'] > CMAX * 0.70) &
    (dados['Temperatura_1'] > temp_media_t1)
]

print(f"Média da Temperatura_1 (T1): {temp_media_t1:.2f}°C")
print(f"Registros com consumo > 70% E temp > média: {len(df_consumo_e_temp)}")

Média da Temperatura_1 (T1): 21.66°C
Registros com consumo > 70% E temp > média: 8


7.1. **Compare os dois DataFrames**

In [ ]:
print("\n--- Comparação entre os DataFrames ---")
print(f"DataFrame Ex. 4 (Apenas Consumo Elevado): {len(df_alto_consumo)} registros")
print(f"DataFrame Ex. 6 (Consumo Elevado + Temp1 > Média): {len(df_consumo_e_temp)} registros")


--- Comparação entre os DataFrames ---
DataFrame Ex. 4 (Apenas Consumo Elevado): 22 registros
DataFrame Ex. 6 (Consumo Elevado + Temp1 > Média): 8 registros


7.2. **explique o efeito da inclusão da temperatura como segundo critério.**

**A inclusão da temperatura como segundo critério reduz o número de registros analisados porque o operador lógico exige o atendimento simultâneo de duas condições. Enquanto o primeiro filtro seleciona isoladamente todos os momentos de consumo extremo de energia, a adição da temperatura interna acima da média elimina os picos de consumo ocorridos em períodos frios ou amenos. Na prática, esse refinamento afunila a amostra para isolar exclusivamente os episódios de alto consumo energético que coincidiram com um ambiente mais quente que o habitual, permitindo à equipe avaliar se o uso intenso dos eletrodomésticos possui relação direta com o aumento da temperatura na residência**

# **Dataset 2 — Steel Industry Energy Consumption (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/851/steel+industry+energy+consumption

**Situação:**

**A equipe de gestão de energia de uma indústria siderúrgica deseja localizar situações de consumo elevado e verificar se esses períodos coincidem com condições de carga elevada ou valores menos favoráveis de fator de potência.**


1.   **Carregue a amostra e renomeie Usage_kWh para Consumo_kWh, simplificando também os nomes dos atributos de fator de potência.**  


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df2 = pd.read_csv("https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET2.csv")

In [ ]:
df2.columns

Index(['Usage_kWh', 'Lagging_Current_Reactive.Power_kVarh',
       'Leading_Current_Reactive_Power_kVarh', 'WeekStatus', 'Day_of_week',
       'Load_Type', 'Leading_Current_Power_Factor',
       'Lagging_Current_Power_Factor'],
      dtype='object')

In [ ]:
df2.rename({
    'Usage_kWh': 'Consumo_kWh',
    'Lagging_Current_Power_Factor': 'Fator_De_Potencia_Atrasada',
    'Leading_Current_Power_Factor': 'Fator_De_Potencia_Capacitiva'
}, axis=1, inplace=True)

In [ ]:
df2.columns

Index(['Consumo_kWh', 'Lagging_Current_Reactive.Power_kVarh',
       'Leading_Current_Reactive_Power_kVarh', 'WeekStatus', 'Day_of_week',
       'Load_Type', 'Fator_De_Potencia_Capacitiva',
       'Fator_De_Potencia_Atrasada'],
      dtype='object')

2.    **Apresente uma inspeção inicial com head(), shape, info() e describe().**


In [ ]:
df2.head()

,Consumo_kWh,Lagging_Current_Reactive.Power_kVarh,Leading_Current_Reactive_Power_kVarh,WeekStatus,Day_of_week,Load_Type,Fator_De_Potencia_Capacitiva,Fator_De_Potencia_Atrasada
0,2.88,3.82,0.0,Weekend,Sunday,Light_Load,100.00,60.20
1,60.77,48.02,0.0,Weekday,Thursday,Maximum_Load,100.00,78.46
2,120.42,59.65,0.0,Weekday,Friday,Maximum_Load,100.00,89.61
3,3.13,0.00,16.6,Weekend,Saturday,Light_Load,18.53,100.00
4,58.86,20.99,0.0,Weekday,Friday,Medium_Load,100.00,94.19


In [ ]:
df2.shape

(7008, 8)

In [ ]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7008 entries, 0 to 7007
Data columns (total 8 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Consumo_kWh                           7008 non-null   float64
 1   Lagging_Current_Reactive.Power_kVarh  7008 non-null   float64
 2   Leading_Current_Reactive_Power_kVarh  7008 non-null   float64
 3   WeekStatus                            7008 non-null   object 
 4   Day_of_week                           7008 non-null   object 
 5   Load_Type                             7008 non-null   object 
 6   Fator_De_Potencia_Capacitiva          7008 non-null   float64
 7   Fator_De_Potencia_Atrasada            7008 non-null   float64
dtypes: float64(5), object(3)
memory usage: 438.1+ KB


In [ ]:
df2.describe()

,Consumo_kWh,Lagging_Current_Reactive.Power_kVarh,Leading_Current_Reactive_Power_kVarh,Fator_De_Potencia_Capacitiva,Fator_De_Potencia_Atrasada
count,7008.000000,7008.000000,7008.000000,7008.000000,7008.000000
mean,27.790164,13.206839,3.803711,84.491448,80.672277
std,33.717719,16.444246,7.356412,30.314798,18.869855
min,2.450000,0.000000,0.000000,12.710000,37.870000
25%,3.200000,2.380000,0.000000,99.710000,63.602500
50%,4.570000,5.040000,0.000000,100.000000,87.990000
75%,52.107500,22.977500,1.857500,100.000000,98.842500
max,153.140000,87.700000,27.650000,100.000000,100.000000


3.  **Determine o maior consumo registrado e calcule um limiar de 75% do consumo máximo.**

In [ ]:
maior_consumo = df2['Consumo_kWh'].max()
print(f"O maior consumo registrado é de {maior_consumo:.2f} kWh.")

O maior consumo registrado é de 153.14 kWh.


In [ ]:
limiar_de_consumo = maior_consumo * 0.75
print(f"O limiar de 75% do consumo máximo é de {limiar_de_consumo:.2f} kWh.")

O limiar de 75% do consumo máximo é de 114.85 kWh.


4. **Crie um DataFrame contendo os registros acima desse limiar e calcule a quantidade e o percentual de registros.**

In [ ]:
df_limiar_consumo = df2[df2['Consumo_kWh'] > limiar_de_consumo]

In [ ]:
print(f"Quantidade de registros acima do limiar: {len(df_limiar_consumo)}")

Quantidade de registros acima do limiar: 129


In [ ]:
print(f"Percentual de registros acima do limiar: {(len(df_limiar_consumo) / df2.shape[0] * 100):.2f}%")

Percentual de registros acima do limiar: 1.84%


5. **Verifique quantos registros desse DataFrame pertencem à categoria Maximum Load.**

In [ ]:
qt_max_load = df_limiar_consumo[df_limiar_consumo['Load_Type'] == 'Maximum_Load']
print(f"Quantidade de registros da categoria Maximum Load no consumo elevado: {len(qt_max_load)}")

Quantidade de registros da categoria Maximum Load no consumo elevado: 62


6. **Escolha um dos atributos de fator de potência, observe seus valores e defina um limite coerente para representar valores mais baixos.**

In [ ]:
df2['Fator_De_Potencia_Atrasada'].describe()

,Fator_De_Potencia_Atrasada
count,7008.000000
mean,80.672277
std,18.869855
min,37.870000
25%,63.602500
50%,87.990000
75%,98.842500
max,100.000000


In [ ]:
limite_fator_potencia_atrasada = 92

7. **Crie um novo DataFrame contendo simultaneamente consumo elevado e fator de potência abaixo do limite escolhido.**

In [ ]:
df_alto_consumo_fp_baixo = df_limiar_consumo[df_limiar_consumo['Fator_De_Potencia_Atrasada'] < limite_fator_potencia_atrasada]

In [ ]:
df_alto_consumo_fp_baixo.shape

(72, 8)

8. **Explique por que esse segundo conjunto de registros pode merecer maior atenção da equipe de energia.**

O DataFrame `df_alto_consumo_fp_baixo` merece maior atenção da equipe de energia porque ele identifica os períodos em que a indústria está enfrentando **alto consumo de energia (Consumo_kWh)** e, simultaneamente, um **fator de potência atrasada abaixo do limite estabelecido**. Essa combinação é crítica por várias razões:

1.  **Custos Elevados**: Alto consumo de energia naturalmente implica em custos operacionais mais altos. Quando isso ocorre junto com um fator de potência baixo, os custos podem ser ainda maiores devido a:
    *   **Perdas na Rede**: Um fator de potência baixo significa que uma corrente elétrica maior é necessária para fornecer a mesma quantidade de potência ativa (útil). Isso causa mais perdas por aquecimento nos condutores e transformadores.
    *   **Multas ou Penalidades**: Muitas concessionárias de energia aplicam multas a consumidores industriais que operam com fator de potência abaixo de um determinado limite (geralmente 0,92 ou 0,95), pois isso sobrecarrega a rede de distribuição.

2.  **Ineficiência Operacional**: Um fator de potência baixo indica que a energia reativa é excessiva em relação à energia ativa, o que representa um uso ineficiente da eletricidade. A equipe de energia pode investigar a causa desse baixo fator de potência (por exemplo, excesso de motores indutivos sem compensação) e implementar soluções.

3.  **Potencial de Otimização**: A identificação desses períodos específicos permite que a equipe concentre seus esforços de otimização. Ao invés de tentar melhorar a eficiência em toda a operação, eles podem focar nos horários ou condições que mais contribuem para o desperdício de energia e custos adicionais. Isso pode incluir a instalação de bancos de capacitores para correção do fator de potência ou ajustes no agendamento de cargas.

Em resumo, esses registros apontam para as situações mais problemáticas em termos de eficiência energética e custos, tornando-os prioritários para análise e intervenção da equipe de gestão de energia.

# **Dataset 3 — Power Consumption of Tetouan City (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/849/power+consumption+of+tetouan+city

**Situação:**

**Uma equipe responsável pelo planejamento da distribuição de energia em Tétouan deseja identificar qual das três zonas apresenta o maior pico de consumo e observar as condições ambientais registradas nos momentos de maior demanda.**

1. **Renomeie as três variáveis de consumo para Consumo_Zona_1, Consumo_Zona_2 e Consumo_Zona_3.**

In [ ]:
df3 = pd.read_csv("https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET3.csv")

In [ ]:
df3 = df3.rename({'Zone 1 Power Consumption': "Consumo_Zona_1",
                  'Zone 2  Power Consumption': "Consumo_Zona_2",
                  'Zone 3  Power Consumption': "Consumo_Zona_3"}, axis = 1)

In [ ]:
df_consumo = df3[['Consumo_Zona_1','Consumo_Zona_2','Consumo_Zona_3']]

In [ ]:
df_consumo.columns

Index(['Consumo_Zona_1', 'Consumo_Zona_2', 'Consumo_Zona_3'], dtype='object')

2. **Determine o consumo máximo registrado em cada uma das três zonas.**

In [ ]:
valor_consumo_maximo = df_consumo.stack().max()
indice_consumo_maximo = df_consumo.stack().idxmax()

# O índice retornado por idxmax será uma tupla: (indice_linha, nome_coluna)
linha_consumo_maximo = indice_consumo_maximo[0]
zona_consumo_maximo = indice_consumo_maximo[1]

3. **Identifique qual zona apresenta o maior pico de consumo na amostra.**

In [ ]:
# Extrai o número da zona a partir do nome da coluna
numero_zona = zona_consumo_maximo.split('_')[-1]

print(f"O pico de consumo (valor: {valor_consumo_maximo:.2f}) pertence à zona: {numero_zona}")

O pico de consumo (valor: 51571.50) pertence à zona: 1


4. **Para a zona identificada, calcule 70% do valor máximo e crie um DataFrame com os registros acima desse limiar.**

In [ ]:
limiar_70 = valor_consumo_maximo * 0.70

In [ ]:
# Cria o novo DataFrame filtrando apenas os registros da zona do pico
df_filtrado = df3[df3[zona_consumo_maximo] > limiar_70]

# Exibe as informações e os primeiros registros do novo DataFrame
print(f"Limiar de 70% do pico ({valor_consumo_maximo:.2f}): {limiar_70:.2f}")
print(f"Total de registros encontrados acima do limiar: {len(df_filtrado)}")
print(df_filtrado[[zona_consumo_maximo]])

Limiar de 70% do pico (51571.50): 36100.05
Total de registros encontrados acima do limiar: 3190
       Consumo_Zona_1
3         42476.93617
8         37286.69202
12        40993.90728
14        37475.16093
19        40676.02649
...               ...
10472     44901.23894
10473     40553.17287
10474     39273.88704
10481     43473.83607
10483     37641.37625

[3190 rows x 1 columns]


5. **Conte os registros e calcule o percentual que eles representam.**

In [ ]:
# Contagem de registros
total_registros = len(df2)
registros_filtrados = len(df_filtrado)

# Cálculo do percentual
percentual = (registros_filtrados / total_registros) * 100

# Exibição dos resultados
print(f"Total de registros na base: {total_registros}")
print(f"Registros acima do limiar: {registros_filtrados}")
print(f"Percentual representado: {percentual:.2f}%")

Total de registros na base: 7008
Registros acima do limiar: 3190
Percentual representado: 45.52%


6. **Calcule a temperatura média e crie um segundo DataFrame contendo consumo acima do limiar e temperatura acima da média.**

In [ ]:
# Calcula a temperatura média da base de dados
temp_media = df3['Temperature'].mean()

# Cria o segundo DataFrame aplicando ambas as condições com o operador lógico & (E)
df_filtrado_duplo = df3[(df3[zona_consumo_maximo] > limiar_70) & (df3['Temperature'] > temp_media)]

# Exibe a média e os resultados obtidos
print(f"Temperatura média: {temp_media:.2f}°C")
print(f"Registros encontrados com ambos os critérios: {len(df_filtrado_duplo)}")
print('')
print(df_filtrado_duplo[[zona_consumo_maximo, 'Temperature']])

Temperatura média: 18.79°C
Registros encontrados com ambos os critérios: 2140

       Consumo_Zona_1  Temperature
12        40993.90728        21.03
14        37475.16093        27.06
19        40676.02649        19.20
21        40256.07103        27.11
25        36511.89369        26.92
...               ...          ...
10468     43020.59016        20.57
10472     44901.23894        23.88
10473     40553.17287        20.25
10474     39273.88704        30.70
10483     37641.37625        26.92

[2140 rows x 2 columns]


7. **Compare os dois conjuntos e explique o que ocorreu com a quantidade de registros após a inclusão da condição ambiental.**

Com a inclusão da condição ambiental (Temperatura > temperatura média), a quantidade de registros diminuiu de `{{registros_filtrados_sem_condicao_ambiental}}` para `{{registros_filtrados_com_condicao_ambiental}}`.

Isso ocorreu porque a segunda condição (temperatura acima da média) agiu como um filtro adicional, removendo os registros que, embora tivessem um consumo elevado, não atendiam ao critério de temperatura. Em outras palavras, o segundo filtro restringiu ainda mais o conjunto de dados, focando apenas nos momentos de alto consumo que também estavam associados a temperaturas mais elevadas.

# **Dataset 4 — Solar Power Generation Data**

**Fonte:** https://www.kaggle.com/datasets/anikannal/solar-power-generation-data | Planta_1

**Situação:**

**Uma equipe de operação de uma usina fotovoltaica deseja localizar períodos de alta geração e identificar quais inversores aparecem com maior frequência nesses momentos. Nesta atividade, o foco inicial será o arquivo de geração da planta, sem realizar junção com o arquivo de sensores.**


1.   **Carregue a amostra e renomeie Usage_kWh para Consumo_kWh, simplificando também os nomes dos atributos de fator de potência.**  


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df4 = pd.read_csv("https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET4.csv")

In [ ]:
df4.columns

Index(['DATE_TIME', 'SOURCE_KEY', 'DC_POWER', 'DAILY_YIELD', 'TOTAL_YIELD',
       'AC_POWER'],
      dtype='object')

In [ ]:
df4.rename({
    "DATE_TIME" : "DATA_HORA",
    "SOURCE_KEY" : "ID",
    "DC_POWER" : "POTENCIA_CC",
    "DAILY_YIELD" : "GERACAO_DIARIA",
    "TOTAL_YIELD" : "GERACAO_TOTAL",
    "AC_POWER" : "POTENCIA_CA"
}, axis=1, inplace=True)

2.    **Apresente uma inspeção inicial com head(), shape, info() e describe().**



In [ ]:
df4.head()

,DATA_HORA,ID,POTENCIA_CC,GERACAO_DIARIA,GERACAO_TOTAL,POTENCIA_CA
0,10-06-2020 19:00,uHbuxQJl8lW7ozc,0.00,6645.000,7240508.000,0.000
1,28-05-2020 00:15,rGa61gmuvPhdLxV,0.00,0.000,7206549.000,0.000
2,17-05-2020 15:15,sjndEbLyjtCKgGv,7333.88,6442.000,7036055.000,718.062
3,01-06-2020 04:15,ih0vzX44oOqAx2f,0.00,0.000,6308232.000,0.000
4,17-05-2020 09:00,z9Y9gH1T5YWrNuG,7665.50,961.375,7021701.375,750.438


In [ ]:
df4.shape

(13756, 6)

In [ ]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13756 entries, 0 to 13755
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   DATA_HORA       13756 non-null  object 
 1   ID              13756 non-null  object 
 2   POTENCIA_CC     13756 non-null  float64
 3   GERACAO_DIARIA  13756 non-null  float64
 4   GERACAO_TOTAL   13756 non-null  float64
 5   POTENCIA_CA     13756 non-null  float64
dtypes: float64(4), object(2)
memory usage: 644.9+ KB


In [ ]:
df4.describe()

,POTENCIA_CC,GERACAO_DIARIA,GERACAO_TOTAL,POTENCIA_CA
count,13756.000000,13756.000000,1.375600e+04,13756.000000
mean,3135.848662,3319.235216,6.978829e+06,306.695495
std,4023.069619,3142.249927,4.162164e+05,393.124284
min,0.000000,0.000000,6.183645e+06,0.000000
25%,0.000000,0.000000,6.509779e+06,0.000000
50%,370.071500,2730.425000,7.146218e+06,35.771400
75%,6350.562500,6293.782500,7.267996e+06,622.065500
max,14416.100000,9163.000000,7.846821e+06,1405.590000


3.  **Determine a maior potência CA registrada.**

In [ ]:
maior_potencia = df4["POTENCIA_CA"].max()
print(f"A maior Potência de Corrente Alternada encontrada foi de {maior_potencia:.2f}kW")

A maior Potência de Corrente Alternada encontrada foi de 1405.59kW


**4.	Calcule um limiar correspondente a 70% da potência CA máxima.**

In [ ]:
limiar_consumo = maior_potencia * 0.70
print(f"O Limiar de consumo definido é de {limiar_consumo:.2f}KW")

O Limiar de consumo definido é de 983.91KW


**5.	Crie um DataFrame contendo somente registros acima desse limite e calcule sua quantidade e percentual.**

In [ ]:
df4_limiar = df4[df4["POTENCIA_CA"] > limiar_consumo]

quantidade_registros = len(df4_limiar)
percentual_registros = (quantidade_registros / df4.shape[0]) * 100

print(f"Quantidade de registros acima do limiar: {quantidade_registros}")
print(f"Percentual de registros acima do limiar: {percentual_registros:.2f}%")


Quantidade de registros acima do limiar: 1185
Percentual de registros acima do limiar: 8.61%


**6.	Utilize value_counts() em SOURCE_KEY dentro do DataFrame de alta geração.**

In [ ]:
df4_limiar['ID'].value_counts()

,count
ID,
adLQvlD726eNBSB,68
McdE0feGgRqW7Ca,65
ZnxXDlPa8U1GXgE,63
pkci93gMrogZuBj,62
ZoEaEvLYb1n2sOq,61
1IF53ai7Xc0U56Y,59
YxYtjZvoooNbGkE,58
sjndEbLyjtCKgGv,57
z9Y9gH1T5YWrNuG,56


**7.	Identifique qual inversor aparece com maior frequência nos registros de alta geração e descreva o que esse resultado permite observar, sem concluir desempenho ou falha apenas com esse recorte**

In [ ]:
inversor_max = df4_limiar['ID'].value_counts().idxmax()
print(inversor_max)

adLQvlD726eNBSB



**8.   Interpretação**

Padronização e Integridade dos Dados: A estrutura do conjunto de dados permitiu a validação dos registros operacionais da planta. A inspeção inicial confirmou a consistência dos atributos essenciais — como a geração diária, total e as potências de corrente contínua (POTENCIA_CC) e alternada (POTENCIA_CA) —, viabilizando o cálculo de métricas de desempenho sem a necessidade de fusão imediata com os dados meteorológicos.  Comportamento da Geração e Operação em Pico: A identificação do valor máximo de potência alternada e a aplicação do limiar de 70% permitiram isolar a janela operacional em que a usina opera em sua faixa de alta produção. O percentual de registros contidos nesse subset revela a frequência com que o sistema atinge altos níveis de conversão de energia, refletindo os períodos de maior disponibilidade de radiação solar ao longo do tempo amostrado.  Distribuição e Frequência dos Inversores: A contagem de frequência dos equipamentos (ID) no subset de alta geração destacou o inversor que registrou o maior volume de picos operacionais. Esse resultado indica qual equipamento esteve mais presente nos momentos de maior entrega de potência da planta.  Limitações Interpretativas: Embora um inversor específico lidere em volume de registros no topo de produção, este achado não permite diagnosticar superioridade técnica ou falha operacional dos demais equipamentos. Como a análise considerou apenas o arquivo de geração, discrepâncias na contagem podem decorrer de variações de insolação local, sombreamento pontual nas strings, diferenças físicas de inclinação/orientação ou eventuais lacunas de telemetria. Para diagnósticos conclusivos de eficiência ou degradação, faz-se necessária a futura integração com os dados dos sensores ambientais.  





# **Dataset 5 — Wind & Solar Energy Production (Kaggle)**

**Fonte:** https://www.kaggle.com/datasets/ahmeduzaki/wind-and-solar-energy-production-dataset

**Situação:**

**Um operador de um portfólio de fontes renováveis deseja comparar a ocorrência de períodos de alta produção solar e alta produção eólica. Como as duas fontes possuem escalas próprias, cada uma deverá ser comparada com o seu próprio valor máximo.**

1. **Renomeie as variáveis principais para Geracao_Solar e Geracao_Eolica**

In [ ]:
df5 = pd.read_csv('https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET5.csv')

In [ ]:
# Filtragem e renomeação das variáveis/DataFrames principais conforme solicitado
Geracao_Solar = df5[df5['Source'] == 'Solar'].copy()
Geracao_Eolica = df5[df5['Source'] == 'Wind'].copy()

print(f"Registros de Geração Solar: {len(Geracao_Solar)}")
print(f"Registros de Geração Eólica: {len(Geracao_Eolica)}")

Registros de Geração Solar: 9378
Registros de Geração Eólica: 42484


2. **Determine o valor máximo registrado para cada fonte**

In [ ]:
solar_max = Geracao_Solar['Production'].max()
eolica_max = Geracao_Eolica['Production'].max()

print(f'O valor máximo registrado para a geração solar é: {solar_max:.2f} MWh')
print(f'O valor máximo registrado para a geração eólica é: {eolica_max:.2f} MWh')

O valor máximo registrado para a geração solar é: 16578.00 MWh
O valor máximo registrado para a geração eólica é: 23446.00 MWh


3. **Calcule separadamente 70% do máximo solar e 70% do máximo eólico**

In [ ]:
solar_max_70 = solar_max * 0.70
eolica_max_70 = eolica_max * 0.70

print(f'70% do valor máximo solar é: {solar_max_70:.2f} MWh')
print(f'70% do valor máximo eólico é: {eolica_max_70:.2f} MWh')

70% do valor máximo solar é: 11604.60 MWh
70% do valor máximo eólico é: 16412.20 MWh


4. **Crie um DataFrame para registros de alta geração solar e outro para registros de alta geração eólica**

In [ ]:
df_alta_geracao_solar = Geracao_Solar[Geracao_Solar['Production'] > solar_max_70]
df_alta_geracao_eolica = Geracao_Eolica[Geracao_Eolica['Production'] > eolica_max_70]

print("DataFrames de alta geração criados com sucesso.")

DataFrames de alta geração criados com sucesso.


5. **Conte os registros de cada DataFrame e calcule seus respectivos percentuais**

In [ ]:
# Contagem de registros
qtd_alta_solar = len(df_alta_geracao_solar)
qtd_total_solar = len(Geracao_Solar)

qtd_alta_eolica = len(df_alta_geracao_eolica)
qtd_total_eolica = len(Geracao_Eolica)

# Cálculo dos percentuais relativas ao total de cada fonte
pct_alta_solar = (qtd_alta_solar / qtd_total_solar) * 100
pct_alta_eolica = (qtd_alta_eolica / qtd_total_eolica) * 100

print(f"Geração Solar Alta (> {solar_max_70:.2f} MWh): {qtd_alta_solar} registros ({pct_alta_solar:.2f}% do total de registros solares)")
print(f"Geração Eólica Alta (> {eolica_max_70:.2f} MWh): {qtd_alta_eolica} registros ({pct_alta_eolica:.2f}% do total de registros eólicos)")

Geração Solar Alta (> 11604.60 MWh): 167 registros (1.78% do total de registros solares)
Geração Eólica Alta (> 16412.20 MWh): 1018 registros (2.40% do total de registros eólicos)


6. **Compare qual fonte aparece com maior frequência acima de 70% do seu próprio máximo**

In [ ]:
if pct_alta_solar > pct_alta_eolica:
    print(f"A fonte SOLAR aparece com maior frequência relativa ({pct_alta_solar:.2f}% vs {pct_alta_eolica:.2f}%).")
elif pct_alta_eolica > pct_alta_solar:
    print(f"A fonte EÓLICA aparece com maior frequência relativa ({pct_alta_eolica:.2f}% vs {pct_alta_solar:.2f}%).")
else:
    print(f"Ambas as fontes apresentam a mesma frequência relativa ({pct_alta_solar:.2f}%).")

A fonte EÓLICA aparece com maior frequência relativa (2.40% vs 1.78%).


7. **Explique por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável.**

### **Explicação Técnica**

**Por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável?**

Não é adequado aplicar um limite absoluto fixo (ex: um valor arbitrário em MW ou MWh) para ambas as fontes porque a energia solar e a energia eólica possuem **perfis de geração, fatores de capacidade e capacidades instaladas totalmente distintos**:

1. **Intermitência e Ciclo Diurno:** A energia solar opera dentro de uma janela restrita (horas de sol), atingindo picos concentrados perto do meio-dia e caindo a zero durante a noite. A energia eólica, por sua vez, pode gerar continuamente ao longo das 24 horas, variando de acordo com regimes de ventos regionais.
2. **Diferenças de Escala e Capacidade:** Se uma usina ou parque eólico tiver uma capacidade máxima pico instalada significativamente maior (ou menor) do que o parque solar, utilizar um único corte numérico distorceria a análise. Um valor alto demais poderia fazer parecer que a fonte de menor escala nunca atinge picos de alta eficiência, enquanto um valor baixo demais classificaria quase toda a produção da fonte de maior escala como 'alta'.

Ao normalizar pelo limite relativo de 70% da capacidade máxima individual de cada fonte, avalia-se o comportamento operacional e o desempenho relativo de cada tecnologia em relação ao seu próprio potencial.

# **Dataset 6 — Individual Household Electric Power Consumption (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/235/individual+household+electric+power+consumption

**Situação:**

**Uma residência possui monitoramento elétrico detalhado e deseja identificar episódios de demanda elevada que também apresentem corrente acima do comportamento médio. O dataset é o mesmo utilizado como referência em aula, mas o critério de análise será diferente.**

1. **Carregue a nova amostra e organize os atributos utilizando nomes mais simples.**

In [ ]:
df6 = pd.read_csv('https://raw.githubusercontent.com/Temitope-Ogunbanjo/Files/main/SERS_CP4_DATASET6.csv')

In [ ]:
df6 = df6.rename({
                      'Global_active_power' : 'Potencia_Ativa',
                      'Global_reactive_power' : 'Potencia_Reativa',
                      'Voltage' : 'Tensao',
                      'Global_intensity' : 'Corrente',
                      'Sub_metering_1' : 'Consumo_1',
                      'Sub_metering_2': 'Consumo_2',
                      'Sub_metering_3': 'Consumo_3'}, axis=1)

2. **Determine o valor máximo de potência ativa.**

In [ ]:
PMAX = df6['Potencia_Ativa'].max()
print(f'O valor máximo de demanda energética é: {PMAX:.2f} kW')

O valor máximo de demanda energética é: 10.54 kW


3. **Calcule 75% do valor máximo e crie um DataFrame contendo registros acima desse limite.**

In [ ]:
P75 = PMAX * 0.75
df_acima_PMAX = df6[df6['Potencia_Ativa'] > P75]

4. **Calcule a quantidade e o percentual de registros selecionados.**

In [ ]:
print(f"Existem cerca de {df_acima_PMAX.shape[0]} de consumidores acima de P75")
percentual = df_acima_PMAX.shape[0] / df6.shape[0] * 100
print(f"O percentual de consumidores acima de P75 é: {percentual:.3f}%")

Existem cerca de 58 de consumidores acima de P75
O percentual de consumidores acima de P75 é: 0.028%


5. **Calcule a corrente média da amostra.**

In [ ]:
corrente_media = df6['Corrente'].mean()
print(f'A corrente média da amostra é: {corrente_media:.2f} A')

A corrente média da amostra é: 4.63 A


6. **Crie um segundo DataFrame contendo simultaneamente potência ativa acima de 75% do máximo e corrente acima da média.**

In [ ]:
df_P_Ativa_Acima = df6[(df6['Potencia_Ativa'] > P75) & (df6['Corrente'] > corrente_media)]

7. **Compare os dois conjuntos e explique o efeito da inclusão da corrente como segunda condição.**

In [ ]:
print(f"Existem cerca de {df3.shape[0]} registros com Potencia Ativa acima de P75 E Corrente acima da média.")
percentual_df3 = df3.shape[0] / df6.shape[0] * 100
print(f"O percentual desses registros é: {percentual_df3:.3f}%")

Existem cerca de 10484 registros com Potencia Ativa acima de P75 E Corrente acima da média.
O percentual desses registros é: 5.116%


A inclusão da condição de `Corrente` acima da média (`corrente_media`) **não resultou em redução no número de registros selecionados** neste caso específico.

Originalmente, com apenas a condição de `Potencia_Ativa` > `P75`, tínhamos **58 registros**, representando **0.028%** da amostra total.

Com a adição da condição `Corrente` > `corrente_media`, o número de registros permaneceu em **58**, representando **0.028%** da amostra.

Isso demonstra que, para este conjunto de dados e os limites estabelecidos, **todos os episódios de demanda elevada (Potência Ativa > P75) também apresentam uma corrente acima do comportamento médio**, o que significa que a segunda condição não filtrou registros adicionais.